### ISSUE TITLE
Question about the true anomaly calculation

### ISSUE THREAD

**Issue by emprice** [lkreidberg/batman#66]
Hi Laura! I was running some benchmark tests and have a question about how batman computes the true anomaly for eccentric orbits. My understanding is that you get mean anomaly from time and orbital period, then eccentric anomaly from mean anomaly, then true anomaly from eccentric anomaly. But when I tried to compute it myself using scipy, I'm getting very different results. Have I misunderstood the calculation? Do you have any insights here?

Here's the code I wrote to compare the two:
```python
import batman                                                                                                                                              
import numpy as np
import scipy.optimize as opt
import matplotlib.pyplot as plt

t0 = 0.
Porb = 2 * np.pi
ecc = 0.05
omega_deg = -90.

nt = 10000
time = np.linspace(-0.3, 0.3, nt)

batp = batman.TransitParams()
batp.t0 = t0
batp.per = Porb
batp.rp = 0.1
batp.a = 5.
batp.inc = 80.
batp.ecc = ecc
batp.w = omega_deg
batp.u = [0.1, -0.02]
batp.limb_dark = 'quadratic' 

model = batman.TransitModel(batp, time, fac=1e-4)
batman_trueanom = model.get_true_anomaly() 
batman_trueanom = np.where(batman_trueanom > 0, batman_trueanom - 2 * np.pi, batman_trueanom)

omega = np.deg2rad(omega_deg)
meananom = 2 * np.pi * (time - t0) / Porb
def gen_eccanom():
    for M in meananom:
        sol = opt.root_scalar(lambda E: (E - ecc * np.sin(E)) - M, x0=M)
        yield sol.root
eccanom = np.array(list(gen_eccanom()))
trueanom = 2 * np.arctan(np.sqrt((1 + ecc) / (1 - ecc)) * np.tan(0.5 * eccanom)) + omega - 0.5 * np.pi
trueanom = np.where(trueanom > 0, trueanom - 2 * np.pi, trueanom)

plt.plot(time, batman_trueanom, lw=2, ls='solid', label='batman')
plt.plot(time, trueanom, lw=2, ls='dashed', label='scipy')
plt.xlabel('time')
plt.ylabel('true anomaly')
plt.legend(loc='best')
plt.show()
```

And the output:
![Figure_1](https://github.com/lkreidberg/batman/assets/3172578/388a0760-169c-4788-af49-4ab7f2b5ece3)

**Comment by lkreidberg** [comment]
Hi Ellen, gosh I haven't thought about this in a long time! to calculate the true anomaly I'm following the procedure outlined in Section 3 of this [book chapter](https://ui.adsabs.harvard.edu/abs/2010exop.book...15M/abstract), and it's implemented in this [routine](https://github.com/lkreidberg/batman/blob/master/c_src/_rsky.c) within batman. I don't have the bandwidth to chase down the difference in detail but hopefully this gives some clarity about what I'm doing. I hope this isn't a bug!!

**Comment by emprice** [comment]
It's not a bug! I finally figured it out, by comparing to the batman source code and a _lot_ of trial and error. Mean anomaly should be calculated relative to the time of periastron (like you are), and I was using time of inferior conjunction. [This ref](https://arxiv.org/pdf/1206.5798) has a lot of details, too, and helped me track down the issue in my code. Now, I'm getting the exact same result as batman. Sorry for the spurious issue! Closing now.
